controllers/user.controller.js

```diff
@@ -0,0 +1,247 @@
+                                            if (err) {
+                                                throw err;
+                                            } else {
+                                                Token.generateToken(userID)
+                                                .then(function(value) {
+                                                    return res.status(200).json({
+                                                        token: value,
+                                                        userID: userID
+                                                    });
+                                                }).catch(function(e) {
+                                                    console.log(e);
+                                                });
+                                            }
+                                        });
+                                    }
+                                });
+                            } catch (err) {
+                                return res.status(505).json({ message: error.message });
+                            }
+                        }
+                    });
+                } else if (results[0].userID) {
+                    return res.status(400).json({ message: "Mail already exist" });
+                }
+            });
+    } catch (error) {
+        return res.status(505).json({ message: error.message });
+    }
+};
+
+let userLogout = async function (req, res) {
+    if (!req.body) {
+        return res.sendStatus(400).json({
+            message: "Empty data!"
+        });
+    }
+
+    const userID = req.body.userID;
+    try {
+        user.query('SELECT token FROM mailPass where userID = ?', userID,
+            function (err, result) {
+                if (err) {
+                    throw err;
+                } else if (!result[0]) {
+                    return res.status(505).json({ message: "Database problem" });
+                } else if (result[0].token) {
+                    const token = result[0].token;
+                    if (token == req.body.token) {
+                        user.query(`UPDATE mailPass SET token = NULL WHERE userID = ?`,
+                            userID, (error, results) => {
+                                if (error) {
+                                    throw error;
+                                } else {
+                                    return res.status(200).json({ message: "By!!!" });
+                                }
+                        });
+                    }
+                } 
+        });
+    } catch (err) {
+        return res.status(400).json({ message: err.message });
+    }
+}
+
+let getUserInfo = async function (req, res) {
+    const userID = req.query.userID;
+    const token = req.query.token;
+    if (userID && token) {
+        user.query(`SELECT * FROM mailPass WHERE userID = ? AND token = ?`,
```

Comment:
things like this should be implemented as route middleware.
For more info please see following:
https://expressjs.com/en/guide/using-middleware.html